src/components/ArtifactsList/ArtifactsList.tsx

```diff
@@ -0,0 +1,257 @@
+      filterValues: ['STANDARD', 'SERVICE'],
+      operator: '==',
+    },
+    filename: {
+      id: 'filename',
+      title: 'Filename',
+      placeholder: 'string | !string | s?ring | st*ng',
+      operator: '=like=',
+    },
+    'targetRepository.repositoryType': {
+      id: 'targetRepository.repositoryType',
+      title: 'Repository Type',
+      filterValues: ['MAVEN', 'GENERIC_PROXY', 'NPM', 'COCOA_POD', 'DISTRIBUTION_ARCHIVE'],
+      operator: '==',
+    },
+    md5: {
+      id: 'md5',
+      title: 'md5',
+      placeholder: 'string | !string | s?ring | st*ng',
+      operator: '=like=',
+    },
+    sha1: {
+      id: 'sha1',
+      title: 'sha1',
+      placeholder: 'string | !string | s?ring | st*ng',
+      operator: '=like=',
+    },
+    sha256: {
+      id: 'sha256',
+      title: 'sha256',
+      placeholder: 'string | !string | s?ring | st*ng',
+      operator: '=like=',
+    },
+  },
+};
+
+const sortOptions: ISortOptions = {
+  identifier: {
+    id: 'identifier',
+    title: 'Identifier',
+    tableColumnIndex: 0,
+  },
+  'build.submitTime': {
+    id: 'build.submitTime',
+    title: 'Build Submit Time',
+    tableColumnIndex: 1,
+  },
+  artifactQuality: {
+    id: 'artifactQuality',
+    title: 'Artifact Quality',
+    tableColumnIndex: 2,
+  },
+  buildCategory: {
+    id: 'buildCategory',
+    title: 'Build Category',
+    tableColumnIndex: 3,
+  },
+  filename: {
+    id: 'filename',
+    title: 'File Name',
+    tableColumnIndex: 4,
+  },
+};
+
+const spaceItemsLg: FlexProps['spaceItems'] = { default: 'spaceItemsLg' };
+
+interface IArtifactsListProps {
+  serviceContainerArtifacts: IServiceContainer;
+  componentId: string;
+}
+
+/**
+ * Component displaying list of Artifacts.
+ *
+ * @param serviceContainerArtifacts - Service Container for Artifacts
+ * @param componentId - Component ID
+ */
+export const ArtifactsList = ({ serviceContainerArtifacts, componentId }: IArtifactsListProps) => {
+  const { getSortParams } = useSorting(sortOptions, componentId);
+
+  const [isArtifactIdentifierParsed, setIsArtifactIdentifierParsed] = useState<boolean>(false);
+
+  const [expandedArtifacts, setExpandedArtifacts] = useState<string[]>([]);
+  const setArtifactExpanded = (artifact: Artifact, isExpanding = true) =>
+    setExpandedArtifacts((prevExpanded) => {
+      const otherExpandedArtifactIdentifiers = prevExpanded.filter((r) => r !== artifact.identifier);
+      return isExpanding ? [...otherExpandedArtifactIdentifiers, artifact.identifier] : otherExpandedArtifactIdentifiers;
+    });
+  const isArtifactExpanded = (artifact: Artifact) => expandedArtifacts.includes(artifact.identifier);
+
+  useEffect(() => {
+    const shouldParse = window.localStorage.getItem('artifacts-list-identifiers-parsed') === 'true';
+    setIsArtifactIdentifierParsed(shouldParse);
+  }, []);
+
+  return (
+    <>
+      <Toolbar>
+        <ToolbarItem>
+          <Filtering filterOptions={filterOptions} componentId={componentId} />
```

Comment:
Improvements(better to have): automatically expand the list item when filtering by `md5`, `sha1`, and  `sha256`. This is because now we no longer show these fields and folds them by default, user would be more clear if they can see the fields they are filtering with. 
![image](https://user-images.githubusercontent.com/6870455/223609024-b7f7803a-e65d-4dd3-bc00-c90247cf3a98.png)

Reply:
Good idea, but will not be part of this Jira, could you please create new Jira for it. Thanks.
Problem currently is with detecting sorting by specific column.

Reply:
NCL-7667 was created for this, thanks, Patrik!